### PR TITLE
Added a function for tracking request result on additional keys, after successful fetch…

### PR DESCRIPTION
**Pull request by WillooWisp**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Possible to use new method for tracking request result on additional keys.

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/requests/src/lib/requests-result.ts [hunk 1]
```diff
@@ -186,9 +186,10 @@ interface Options {
   cacheResponseData?: boolean;
 }
 
-export function trackRequestResult<TData>(
+export function trackManyRequestResult<TData>(
   key: unknown[],
-  options?: Options
+  additionalKeysFn?: (_: TData) => unknown[][],
```
Comment (NetanelBasal) [inline packages/requests/src/lib/requests-result.ts:191]: I prefer one function:

```ts
trackRequestResult(['todos'], {
  additionalKeys: todos => todos.map(todo => (['todos', todo.id]))
})
```

File: packages/requests/src/lib/requests-result.spec.ts [hunk 2]
```diff
@@ -653,4 +655,41 @@ describe('requests result', () => {
...
+      return timer(1000).pipe(
+        map(() => createTodo(id)),
+        tap((todo) => store.update(upsertEntities([todo]))),
+        tap(todo => spy(todo)),
+        trackRequestResult(['todos', id])
+      );
+    }
+
+    // Todos must have been fetched before caching can be done.
+    timer(1000).subscribe(() => getTodo(1).subscribe());
```
Comment (NetanelBasal) [inline packages/requests/src/lib/requests-result.spec.ts:689]: Add a test with joinRequestResult for todo id 1 and check the response and the cached response with the number of times it's called

File: packages/requests/src/lib/requests-result.ts [hunk 3]
```diff
@@ -272,6 +273,13 @@ export function trackRequestResult<TData>(
               }
 
               updateRequestResult(key, newResult);
+
+              if (additionalKeysFn) {
+                const mainKey = resolveKey(key);
+                for (const keys of additionalKeysFn(sourceData)) {
```
Comment (NetanelBasal) [inline packages/requests/src/lib/requests-result.ts:279]: Why you are not using the updateRequestResult function?

Comment (WillooWisp) [reply packages/requests/src/lib/requests-result.ts:279]: I did at first, but realized that it does nothing if nothing is cached for the key, and I simply want to cache the same request result for another key.

File: packages/requests/src/lib/requests-result.spec.ts (hunk 2 above)
Comment (WillooWisp) [reply packages/requests/src/lib/requests-result.spec.ts:689]: New test has been added, not sure if it covers what you meant though.

**Comment by NetanelBasal** [comment]
One last thing, your commits are not based on our guidelines. Please squash your commits and use `npm run c`

**Comment by WillooWisp** [comment]
> One last thing, your commits are not based on our guidelines. Please squash your commits and use `npm run c`

Commits squashed, but unable to run git-gz, does not find git whatever I do, not on Windows and not in WSL.

**Comment by WillooWisp** [comment]
Squashed commits and npm run c used to create proper commit.

**Comment by NetanelBasal** [comment]
Tests failed

**Comment by WillooWisp** [comment]
> Tests failed

Fixed!

**Comment by WillooWisp** [comment]
@NetanelBasal anything else stopping the PR acceptance? When can we expect a release including this?

**Pull request merged**